woocommerce/woocommerce-ios
# Simplified login: Prologue screen updates

Closes: #7997

- [ ] Please review - https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/695

### Description

For the first iteration of the Simplified Login flow, we need to update the prologue screen.

This PR attempts to update the prologue screen for the various combination based on the following feature flags. (Please note that these feature flags will be replaced with A/B testing eventually)
- `simplifiedLoginFlowI1`
- `storeCreationMVP`

**Changes**
- Remove "New to WooCommerce" button from prologue by turning off `newToWooCommerceLinkInLoginPrologue` feature flag.
- Add a OR divider and show buttons based on the feature flags. (Changes in the Authenticator library)
- Rename create store button
- Use the same copy for the intro text ("WooCommerce is a customizable...") for all variants

### Testing instructions
- Install and launch the app
- Skip onboarding to land in login prologue screen
- Verify the following based on the feature flags `simplifiedLoginFlowI1` and `storeCreationMVP`
  - You can configure those feature flags from `DefaultFeatureFlagService.isFeatureFlagEnabled()` method

1. Simplified login ON
    - Site creation ON
      - "Login" and "Get Started" buttons are visible.
      - Tapping "Login" should start the WPCOM login process.
      - Tapping "Get Started" should start account creation process.
    - Site Creation OFF
      - "Login" button visible.
      - Tapping "Login" should start the WPCOM login process.
2. Simplified login OFF
    - Site creation ON
      - "Continue with WordPress.com", "Enter Your Store Address" and "Get Started" buttons are visible. (Ref screenshots section)
      - "Continue with WordPress.com" and "Enter Your Store Address" should behave as existing buttons. 
      - Tapping "Get Started" should start account creation process.
    - Site Creation OFF
      - "Continue with WordPress.com" and "Enter Your Store Address" visible and they should behave as existing buttons. 

**Feature carousel removed in all variations**
- Observe that there is no scrollable feature carousel present. 
- Only one screen with intro text ("WooCommerce is a customizable...") should be present in the login carousel. (Even after relaunching the app.)

### Screenshots

| | Simplified login ON | Simplified login OFF |
| --- | --- | --- |
| **Site creation ON** | ![BothSimplifiedLoginAndSiteCreationTurnedOn](https://user-images.githubusercontent.com/524475/199228170-5a114436-1f52-41e8-97eb-c80e01cc5201.png) | ![OnlySiteCreationOn](https://user-images.githubusercontent.com/524475/199228182-cee4f20a-d22e-48bc-b59f-fa1097d2b36c.png) |
| **Site creation OFF** | ![OnlySimplifiedLoginOn](https://user-images.githubusercontent.com/524475/199228176-f2f0a952-925c-42e2-9df3-b5e597120f1a.png) | ![BothSimplifiedLoginAndSiteCreationTurnedOff](https://user-images.githubusercontent.com/524475/199228164-5cd07fa7-b861-46c5-afbf-931adae4f39f.png) |

---
